conda/recipes/ucxx/meta.yaml

```
@@ -59,7 +64,10 @@ outputs:
       host:
         - ucx
       run:
+        {% if cuda_major == "11" %}
         - cudatoolkit {{ cuda_spec }}
+        {% endif %}
+        - cuda-version {{ cuda_spec }}
```

In some cases we've specified `- cuda-version ={{ cuda_version }}` in `host` and then used `pin_compatible` in `run` to generate the min/max spec. [Example from cudf here](https://github.com/rapidsai/cudf/blob/aed7174eae6c6eb38fbf186938df44f88787cf29/conda/recipes/cudf/meta.yaml#L98). Do we want to do that here?

---

I'm actually now thinking if we should have this in `run` at all. CUDA is not a required dependency, so I believe we should even remove this.

---

This is tricky. If CUDA is not a hard dependency, it shouldn't be in `run`. Yet if the optional CUDA bits differ between the two builds, I think we still need two package variants, one constrained to CUDA 11 and one constrained to CUDA 12. So maybe we need to put `cuda-version` in `run_constrained` so that it isn't _installed_, but limits that ucxx build to a particular CUDA major version?

---

(We discussed offline and CUDA _is_ a hard dependency. The comment below happens to be on the same line as the previous discussion thread but this comment is no longer related to the thread above.)

This will use the pinning from `host` and allow any compatible same-major version.
```suggestion
        - {{ pin_compatible('cuda-version', max_pin='x', min_pin='x') }}
```